server/src/test/java/org/elasticsearch/rest/action/RestActionTestCase.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest.action;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.indices.breaker.NoneCircuitBreakerService;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.FakeRestChannel;
+import org.elasticsearch.usage.UsageService;
+
+import java.util.Collections;
+
+import static org.mockito.Mockito.mock;
+
+public abstract class RestActionTestCase extends ESTestCase {
+    private RestController controller;
+
+    public void setUp() throws Exception {
+        super.setUp();
+        controller = new RestController(Collections.emptySet(), null,
```

Comment:
Are you imagining a comment like 'these are effectively final, but can't be marked as such because they're initialized in the `@Before` method`?

Reply:
Could you declare the controller as final? I *think* it looks like it'd work.

Reply:
I caught up with @nik9000 offline and he is okay sticking with the style where we use `@Before` methods (and instance variables aren't final), instead of switching everything over to instance variables/ constructors. The reasoning is that there is an existing convention to use `@Before` methods in JUnit tests as opposed to standard object construction.

Reply:
I tend to use `@Before` as little as possible because it looks to me like a signal that I'm doing something complex. I see that this might be out of step with normal maven usage though. So I'm totally fine not doing it.